### ISSUE TITLE
CLI input and ignoreFiles glob weirdness

### ISSUE THREAD

**Issue by jaydenseric** [stylelint/stylelint#1769]
A bug where CLI input globs only work as expected when wrapped in quotes.

With this directory structure (`test.css` files contain deliberate errors):

```
package.json
test.css
test/test.css
test/test/test.css
bundle/bundle.css
```

With the following stylelint config in `package.json`:

```json
{
  "stylelint": {
    "extends": "stylelint-config-standard",
    "ignoreFiles": "bundle/**"
  }
}
```

Running:

```shell
stylelint **/*.css
```

Does not find all errors.

Running:

```shell
stylelint '**/*.css'
```

Does.

This sort of weirdness has been happening for some time and is present in `v7.1.0`.

Removing the `ignoreFiles` config allows the errors to be found. Keeping that config, but deleting `bundle/bundle.css` also allows the errors to be found. Keeping the ignore config but changing the input glob to `**/**/*.css` allows the nested errors to be found, but not in the top level.

Here are online glob tests for the above example:

- [CLI input glob](http://www.globtester.com/#p=eJzT0tLX0ksuLgYAB6QCJQ%3D%3D&r=eJylj0EOgjAQRa%2FSdKWJ0sCqwNC7tKUiWtumU4x6eiEuRGOCxs38zc97f6Dtz0RbidhQKgCDdATT1ZqG7rxLW%2BxvpiJ5mRXhUlMRpD7KzmQH9A7Y1BbARoSAGeckk95%2FAUsGU6YRl0BTjz3L%2F%2BjYG%2BuHD7S3PlYkdmrFiw3JOR9Pydc1%2BSRTg2utYY%2BYu5ZWgoriZdcdmviNAw%3D%3D&)
- [`ignoreFiles` glob](http://www.globtester.com/#p=eJxLKs1LyUnV19ICABE4Av4%3D&r=eJydjs0OwiAQBl%2BFcNJES9oTbbe8C1CsVQTCUqM%2BvW08qI0%2F0cvuZTLzQdsfibYSsaFUAAbpCKazNQ3deJfW2F9MRfIyK8KppiJIvZedyXboHbCJFsBGhYDfPMlgyjTiB8eEsDv3Z4TNNG9yB5n0dqbU3vpYkdipBS9WJOd8PCVf1uRVTA2utYbd3mPr20pQUTztugJRC4bc&)

I only thought to try wrapping the CLI input glob with quotes after running `stylelint -h` and reading:

> If an input argument is wrapped in quotation marks, it will be passed to node-glob for cross-platform glob support.

This is not mentioned in the online [CLI documentation](http://stylelint.io/user-guide/cli).

**Comment by davidtheclark** [comment]
This is not a bug. It's a documented part of the CLI. The online CLI documention just points you to `stylelint --help`, where it is mentioned.

@evilebottnawi sounds like you have a different issue?

**Comment by evilebottnawi** [comment]
@danieldiekmeier hm, need testing if i found error, create issue

**Comment by danieldiekmeier** [comment]
@evilebottnawi Please stop mentioning me. You want davidtheclark, not me.
